diesel/src/expression/sql_literal.rs

```diff
@@ -94,5 +174,70 @@ impl<ST> NonAggregate for SqlLiteral<ST> {}
 /// # }
 /// ```
 pub fn sql<ST>(sql: &str) -> SqlLiteral<ST> {
-    SqlLiteral::new(sql.into())
+    SqlLiteral::new(sql.into(), ())
+}
+
+#[derive(Debug, Clone, Copy)]
+#[must_use = "Queries are only executed when calling `load`, `get_result`, or similar."]
+pub struct UncheckedBind<Query, Value, ST> {
+    query: Query,
+    value: Value,
+    _marker: PhantomData<ST>,
+}
+
+impl<Query, Value, ST> UncheckedBind<Query, Value, ST>
+where
+    Query: Expression,
+{
+    pub fn new(query: Query, value: Value) -> Self {
+        UncheckedBind {
+            query,
+            value,
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn sql(self, sql: &str) -> SqlLiteral<Query::SqlType, Self> {
+        SqlLiteral::new(sql.into(), self)
+    }
+
+    pub fn bind<ST2, U>(self, value: U) -> UncheckedBind<Self, U, ST2> {
+        UncheckedBind::new(self, value)
+    }
+}
+
+impl<Query, Value, ST> Expression for UncheckedBind<Query, Value, ST> {
+    type SqlType = ST;
+}
+
+impl<Query, Value, ST> QueryId for UncheckedBind<Query, Value, ST>
+where
+    Query: QueryId,
+    ST: QueryId,
+{
+    type QueryId = UncheckedBind<Query::QueryId, (), ST::QueryId>;
+
+    const HAS_STATIC_QUERY_ID: bool = Query::HAS_STATIC_QUERY_ID && ST::HAS_STATIC_QUERY_ID;
 }
+
+impl<Query, Value, ST, DB> QueryFragment<DB> for UncheckedBind<Query, Value, ST>
+where
+    DB: Backend + HasSqlType<ST>,
+    Query: QueryFragment<DB>,
+    Value: ToSql<ST, DB>,
+{
+    fn walk_ast(&self, mut out: AstPass<DB>) -> QueryResult<()> {
+        self.query.walk_ast(out.reborrow())?;
+        out.push_bind_param(&self.value)?;
```

Comment:
Rather than duplicating bind param handling here, what do you think about structuring this differently so that it just holds the bind parameter?

```
struct UncheckedBind<Query, Value> {
    query: Query,
    value: Value,
}

impl<ST, T> SqlLiteral<ST, T> {
    pub fn bind<BindST, U>(self, value: U) -> UncheckedBind<Self, U::Expression>
    where
        U: AsExpression<BindST>,
```